include/picongpu/plugins/saxs/Saxs.kernel

```diff
@@ -0,0 +1,291 @@
+
+                ForEachIdx<IdxConfig<1, numWorker>> onlyMaster{workerIdx};
+
+                /* The Master process (thread 0) in every thread block is in
+                 * charge of loading a frame from
+                 * the current super cell and evaluate the total number of
+                 * particles in this frame.
+                 */
+                onlyMaster([&](uint32_t const, uint32_t const) {
+                    np_s = 0;
+                    nmp[0] += particlesInFrame;
+                });
+
+                __syncthreads();
+
+                using ParticleDomCfg = IdxConfig<frameSize, numWorker>;
+
+                // loop over all particles in the frame
+                ForEachIdx<ParticleDomCfg> forEachParticle{workerIdx};
+
+                // load particle positions to shared memory
+                forEachParticle([&](uint32_t const linearIdx, uint32_t const) {
+                    // only threads with particles are running
+                    if (linearIdx < particlesInFrame)
+                    {
+                        auto par = frame[linearIdx];
+                        // calculate global position
+                        lcellId_t const cellIdx = par[localCellIdx_];
+                        // position inside of the cell
+                        floatD_X const pos = par[position_];
+                        // calculate global position of cell
+                        DataSpace<simDim> const globalPos(
+                            superCellOffset +
+                            DataSpaceOperations<simDim>::template map<
+                                SuperCellSize>(cellIdx));
+
+                        /* Add global position of cell with local position
+                         * of particle in cell
+                         */
+
+                        // set z component to zero in case of simDim == DIM2
+                        r_s[linearIdx][2] = 0.0;
+                        // run over all components and compute global position
+                        for (int i = 0; i < simDim; ++i)
+                            r_s[linearIdx][i] =
+                                (float_X(globalPos[i]) + pos[i]) * cellSize[i];
+
+                        /* get macro-particle weighting
+                         *
+                         * Info:
+                         * the weighting is the number of real particles
+                         * described by a macro-particle
+                         */
+                        saxsWeighting_s[linearIdx] = par[weighting_];
+
+                    } // END: only threads with particles are running
+                });
+
+                __syncthreads(); // wait till every thread has loaded its
+                                 // particle data
+
+                // meter -> Angstrom
+                float_64 const meter2angstrom = 1e10;
+
+                // run over all q for this thread
+                for (int idx = workerIdx; idx < n_q; idx += T_numWorkers)
```

Comment:
Suggested change of parallelism (3): remove this line, the loop body stays here with `idx` replaced with `workerIdx`. (now we will just do a single iteration of the old loop).